src/debugAdapter/goDebug.ts

```diff
@@ -1587,6 +1585,70 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	private logDelveError(err: any, message: string) {
+		if (err) {
+			let errorMessage = err.toString();
+			// Handle unpropagated fatalpanic errors with a more user friendly message:
+			// https://github.com/microsoft/vscode-go/issues/1903#issuecomment-460126884
+			// https://github.com/go-delve/delve/issues/852
+			// This affects macOS only although we're agnostic of the OS at this stage, only handle the error
+			if (errorMessage === 'bad access') {
+				errorMessage = 'unpropagated fatalpanic: signal SIGSEGV (EXC_BAD_ACCESS). This fatalpanic is not traceable on macOS, see https://github.com/go-delve/delve/issues/852';
+			}
+			logError(message + ' - ' + errorMessage);
+		}
+		this.dumpStacktrace();
+	}
+
+	private dumpStacktrace() {
+		// Get current goroutine
+		// Debugger may be stopped at this point but we still can (and need) to obtain state and stacktrace
+		let goroutineId = 0;
+		this.delve.call<DebuggerState | CommandOut>('State', [{ NonBlocking: true }], (err, out) => {
```

Comment:
I tested this out on macOS, unfortunately `dlv` doesn't report the correct go routine in this case (I see the same behavior using vscode and dlv) so I think we can remove this.  What it means though is if the panic is on a go routine other than zero you won't get the correct stack trace.  If we still want to include the stack trace then I think we should add a message indicating that it might not be correct.  :(

Reply:
I would leave it just for diagnose purposes with a warning message saying that the stacktrace might differ from the expected goroutine

> I tested this out on macOS, unfortunately dlv doesn't report the correct go routine in this case (I see the same behavior using vscode and dlv)

This is probably related to the original dlv issue, or maybe it is a different issue that might be actionable, nevertheless, that is still something that should be discussed upstream, unfortunately